compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -209,20 +209,18 @@ impl<Prov> Allocation<Prov> {
     ) -> Self {
         let slice: Cow<'a, [u8]> = slice.into();
         let size = Size::from_bytes(slice.len());
-        // TODO come up with a cleaner way to align the box based on the alignment parameter
-        let count128 = (slice.len() / 16) + 1;
-        let mut vec128: Vec<u128> = Vec::with_capacity(count128);
+        let align_usize: usize = align.bytes().try_into().unwrap();
+        let count_align = ((slice.len() / align_usize) + 1)*align_usize;
```

Comment:
While I did something similar in the hack above, you should really only allocate extra space like this if you need to allocate in chunks that don't match the size.

This overallocates 100% of the time, even if the request is a multiple of its alignment.